### ISSUE TITLE
Flux.1 LoRA training issue

### ISSUE THREAD

**Issue by paoloski97** [kohya-ss/sd-scripts#1571]
Hi all, I keep getting `avr_loss=nan` and can't solve the problem. Can anyone help me?

Here is my training command:

`!accelerate launch ./kohya_ss/sd-scripts/flux_train_network.py --pretrained_model_name_or_path /kaggle/input/flux/pytorch/flux.1-dev/1/flux1-dev.safetensors --clip_l /kaggle/input/clip-flux/pytorch/default/3/clip_l.safetensors --t5xxl /kaggle/input/clip-flux/pytorch/default/3/t5xxl_fp16.safetensors --ae /kaggle/input/clip-flux/pytorch/default/3/ae.sft --cache_latents_to_disk --save_model_as safetensors --sdpa --persistent_data_loader_workers --max_data_loader_n_workers 2 --gradient_checkpointing --gradient_accumulation_steps 1 --mixed_precision fp16 --save_precision fp16 --network_module networks.lora_flux --network_dim 64 --learning_rate 3e-5 --network_train_unet_only --cache_text_encoder_outputs --cache_text_encoder_outputs_to_disk --fp8_base --highvram --max_train_epochs 16 --save_every_n_epochs 4 --dataset_config ./dataset/dataset-config.toml --output_dir ./Output --output_name flux-lora-Erin --timestep_sampling sigmoid --model_prediction_type raw --guidance_scale 3.5 --loss_type l2 --optimizer_type adafactor --optimizer_args "relative_step=False" "scale_parameter=False" "warmup_init=False"`

Here is the console output:

```
override steps. steps for 16 epochs is / 指定エポックまでのステップ数: 2400
enable fp8 training for U-Net.
enable fp8 training for Text Encoder.
running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 150
  num reg images / 正則化画像の数: 0
  num batches per epoch / 1epochのバッチ数: 150
  num epochs / epoch数: 16
  batch size per device / バッチサイズ: 1
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 2400
steps:   0%|                                           | 0/2400 [00:00<?, ?it/s]2024-09-06 08:01:28 INFO     text_encoder is not needed    ]8;id=740685;file:///kaggle/working/kohya_ss/sd-scripts/train_network.py\train_network.py]8;;\:]8;id=846258;file:///kaggle/working/kohya_ss/sd-scripts/train_network.py#1029\1029]8;;\
                             for training. deleting to                          
                             save memory.                                       
2024-09-06 08:01:29 INFO     unet dtype:                   ]8;id=911102;file:///kaggle/working/kohya_ss/sd-scripts/train_network.py\train_network.py]8;;\:]8;id=959986;file:///kaggle/working/kohya_ss/sd-scripts/train_network.py#1046\1046]8;;\
                             torch.float8_e4m3fn, device:                       
                             cuda:0                                             

epoch 1/16
/opt/conda/lib/python3.10/multiprocessing/popen_fork.py:66: RuntimeWarning: os.fork() was called. os.fork() is incompatible with multithreaded code, and JAX is multithreaded, so this will likely lead to a deadlock.
  self.pid = os.fork()
huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
To disable this warning, you can either:
	- Avoid using `tokenizers` before the fork if possible
	- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
To disable this warning, you can either:
	- Avoid using `tokenizers` before the fork if possible
	- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
2024-09-06 08:01:30 INFO     epoch is incremented.             ]8;id=258849;file:///kaggle/working/kohya_ss/sd-scripts/library/train_util.py\train_util.py]8;;\:]8;id=160239;file:///kaggle/working/kohya_ss/sd-scripts/library/train_util.py#668\668]8;;\
                             current_epoch: 0, epoch: 1                         
2024-09-06 08:01:30 INFO     epoch is incremented.             ]8;id=135186;file:///kaggle/working/kohya_ss/sd-scripts/library/train_util.py\train_util.py]8;;\:]8;id=209551;file:///kaggle/working/kohya_ss/sd-scripts/library/train_util.py#668\668]8;;\
                             current_epoch: 0, epoch: 1                         
/opt/conda/lib/python3.10/site-packages/torch/utils/checkpoint.py:1399: FutureWarning: `torch.cpu.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cpu', args...)` instead.
  with device_autocast_ctx, torch.cpu.amp.autocast(**cpu_autocast_kwargs), recompute_context:  # type: ignore[attr-defined]
steps:   2%|▎                 | 48/2400 [10:39<8:42:16, 13.32s/it, avr_loss=nan]
```

**Comment by paoloski97** [comment]
Thanks for the quick reply, but is there no way to use `fp16` instead of `bf16`? I am on a free Kaggle Notebook and it does not support bf16.

**Comment by pranauv1** [comment]
@paoloski97 if you don't mind can you share the notebook?

**Comment by paoloski97** [comment]
@pranauv1 Of course I can share the notebook. I am working now, as soon as I get home and have a moment I will fix the notebook (remove unnecessary cells, add useful comments, ecc..) and post it.

**Comment by paoloski97** [comment]
@pranauv1 Here is the link to the pubblic Notebook.

https://www.kaggle.com/code/doz2guyz/flux-1-kohya-lora-training

**Comment by Razbotics** [comment]
@kohya-ss can you explain me why fp16 gives avg loss Nan?

**Comment by Rizwanali324** [comment]
> @pranauv1 Here is the link to the pubblic Notebook.
> 
> https://www.kaggle.com/code/doz2guyz/flux-1-kohya-lora-training

gettting this eror


The error message indicates that the script is failing due to a missing file: dataset/dataset-config.toml.

i have only images and captions file in myfolder. plz share remaining resouces

**Comment by paoloski97** [comment]
@Rizwanali324 I think you have to create the `dataset-config.toml` file for your dataset.

**Comment by Rizwanali324** [comment]
[> Author

/kaggle/input/clip-flux/pytorch/default/3/clip_l.safetensors, resulting in a FileNotFoundError](https://github.com/kohya-ss/sd-scripts/issues/1571#issuecomment-2485715910)

**Comment by paoloski97** [comment]
You need to upload the 'clip_l' on Kaggle, add it to the Notebook and then modify the `--clip_l` argument with the path to the model.

**Comment by Rizwanali324** [comment]
> You need to upload the `clip_l` on Kaggle, add it to the Notebook and then modify the `--clip_l` argument with the path to the model.

i added both clip_l and txx5 after that i run out memory
do u have any recorded video?

**Comment by a442509097** [comment]
After switching to BF16, it worked correctly, but the speed decreased by approximately 8 times compared to FP16.
